rust/rope/src/find.rs

```diff
@@ -163,50 +177,111 @@ fn find_core(cursor: &mut Cursor<RopeInfo>, pat: &str,
 /// Compare whether the substring beginning at the current cursor location
 /// is equal to the provided string. Leaves the cursor at an indeterminate
 /// position on failure, but the end of the string on success.
-pub fn compare_cursor_str(cursor: &mut Cursor<RopeInfo>, mut pat: &str) -> bool {
+pub fn compare_cursor_str(cursor: &mut Cursor<RopeInfo>, mut pat: &str) -> Option<usize> {
+    let start_position = cursor.pos();
     if pat.is_empty() {
-        return true;
+        return Some(start_position);
     }
     let success_pos = cursor.pos() + pat.len();
     while let Some((leaf, pos_in_leaf)) = cursor.get_leaf() {
         let n = min(pat.len(), leaf.len() - pos_in_leaf);
         if leaf.as_bytes()[pos_in_leaf..pos_in_leaf + n] != pat.as_bytes()[..n] {
-            return false;
+            return None;
         }
         pat = &pat[n..];
         if pat.is_empty() {
             cursor.set(success_pos);
-            return true;
+            return Some(start_position);
         }
         let _ = cursor.next_leaf();
     }
-    false
+    None
 }
 
 /// Like `compare_cursor_str` but case invariant (using to_lowercase() to
 /// normalize both strings before comparison).
-fn compare_cursor_str_casei(cursor: &mut Cursor<RopeInfo>, pat: &str) -> bool {
+fn compare_cursor_str_casei(cursor: &mut Cursor<RopeInfo>, pat: &str) -> Option<usize> {
+    let start_position = cursor.pos();
     let mut pat_iter = pat.chars();
     let mut c = pat_iter.next().unwrap();
     loop {
         if let Some(rope_c) = cursor.next_codepoint() {
             for lc_c in rope_c.to_lowercase() {
                 if c != lc_c {
-                    return false;
+                    return None;
                 }
                 if let Some(next_c) = pat_iter.next() {
                     c = next_c;
                 } else {
-                    return true;
+                    return Some(start_position);
                 }
             }
         } else {
             // end of string before pattern is complete
-            return false;
+            return None;
         }
     }
 }
 
+/// Compare whether the substring beginning at the cursor location matches
+/// the provided regular expression. The substring begins at the beginning
+/// of the leave/start of the line.
+/// If the regular expression can match multiple lines then all leaves are
+/// consumed and matched against the regular expression. Otherwise only the
+/// current leave is matched.
```

Comment:
Did you mean `leaf` here? Also on line 228 three lines above.

Reply:
Yes, right. Thanks!